lib/asset/baseAsset.js

```diff
@@ -18,12 +18,15 @@ class BaseAsset {
         return 'EknStaticTag';
     }
 
-    constructor(verifier = assetVerifier) {
+    constructor(metadata = {}, verifier = assetVerifier) {
         this._asset_id = this._newAssetId();
         this._ekn_tags = [];
+        this._verifier = verifier;
         this._tags = [];
 
-        this._verifier = verifier;
+        if (metadata) {
```

Comment:
Are you sure `{}` evaluates to false-y? Since you have a default value for the parameter, is this code path not always taken? Maybe I'm missing something

Reply:
You are right, this is always going through.